### DISCUSSION TITLE
v1.9.0 Release - early access build - please try this and give feedback!

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#640] [Announcements]
<img src="https://user-images.githubusercontent.com/6903107/141677022-a1d6e66d-5920-419c-80cc-0ad12238d92c.png" width="500">

This is the future v1.9.0 release, you can test it by using the [portable build](https://ci.appveyor.com/api/projects/JohnTheGr8/flow-launcher/artifacts/Output%2FPackages%2FFlow-Launcher-Portable.zip?branch=master&pr=true).

We recommend when testing this, to keep it separate from your everyday Flow Launcher because there may be changes to some of the setting files which may not be backwards compatible with the current release. If you are sure you will be using this release going forward, feel free to replace it with your current one as these changes are stable. 

Version of this is marked as the current latest release version so it can still be updated when v1.9.0 come out.

Any issues, let us know below.

**Comment by pc223** [comment]
Did a quick round of tests, seems OK on my machine 👍

**Comment by ags1234** [comment]
I was looking for a launcher and just found your program.

The good
- the interface. The program looks good. Both main search form and options form are pleasant to work with and look at;
- themes are nice;
- plugins system is very useful.

The bad
- you give the link to v1.9.0, but in the archive it is v1.8.3;
- a portable program that modifies my system settings and starts with a check mark at "start on system startup"? No way!
- a launcher that is 313 MB!! This is an insane amount of occupied space for such a program! I would consider using it if it were somewhere between 5-10 MB. FindAndRunRobotPortable v2.239.03 by DonationCoder is 12,3 MB which is more down to this Earth.

**Reply by taooceros** [reply]
Hello, thanks for trying our pre-release version.
> you give the link to v1.9.0, but in the archive it is v1.8.3;

The archive is v1.8.3 because we don't enable small version, so v1.8.3 is kept to enable auto-update to v1.9.0 after releasing.

> a portable program that modifies my system settings and starts with a check mark at "start on system startup"? No way!

Yeah I think it's not reasonable to enable auto-start for portable version.

> a launcher that is 313 MB!! This is an insane amount of occupied space for such a program! I would consider using it if it were somewhere between 5-10 MB. FindAndRunRobotPortable v2.239.03 by DonationCoder is 12,3 MB which is more down to this Earth.

It's large mainly because of runtime and the duplicate package from our installer squirrel. We are trying trimming now,  and the effect is significant but it's not quite possible to get down to 5-10MB. 
~~I have built one version for testing so if you really care about size, try that one. http://sendanywhe.re/MJC3O1VO
It's 50MB after compression but if you don't want auto-update (probably auto-update may still work), just delete the nupkg in the package folder, which should decrease the size by half.~~

**Comment by ags1234** [comment]
Thank you for the reply.

I downloaded the version from your link above. 96 MB after decompression and 71 MB after FlowLauncher-1.0.0-full.nupkg is gone. It is better than 313 MB, but still extraordinarily large for what it does.
Now it asks me to install .NET Framework 6. Why would a launcher need .NET 6? What special components are there, that cannot be found anywhere else? I am not going to install .NET 6 just for this.  This could be done with no .NET framework at all, or could be done with .NET 2.0 and still be compatible with all Windows versions, form XP to 11.

I like your program and I appreciate your effort and the fact that you are making this free and open-source.
I do not want to leave the impression I care about space that much. But I do care about efficiency.

Look at other software: Total Commander, Notepad++, MPC-HC, CDBurnerXP, NIUBI Partition Editor, Victoria, WinRAR, BleachBit, HE Hardware Read Write Utility, AdiIRC, StartIsBack, ProcessExplorer, Process Hacker, Find and Run Robot, are just those that I remember now, but I know there are others. What do they have in common? They are small programs, offering a lot of features. They have become very well regarded in their field because they are well developed, they are efficient, with very few dependencies, if any. Every one of those programs is smaller than half of Flow Launcher. I can only imagine what the user's reaction might be if starting tomorrow, every program from this list is at least 300 MB and could not start without .NET 6. Looking at launchers on Softpedia, I couldn't find a launcher as big as this. On the other hand, there are launchers there that are below 100 KB in size.

LibreOffice 7.2, the latest version as of now, is 600 MB. _An entire office suite in two Flow Launchers!_

That is just my opinion. And I took the time to write it here because I liked Flow Launcher. I would consider using it if it were 5-10 MB. But if it is 86.5 MB compressed (on Softpedia) and 313 MB or 96 MB decompressed, it will have a hard time being used by many users.

**Reply by taooceros** [reply]
Thank you so much for writing  this. We really appreciate it!

However, it's not easy. Most of the program you describe are written in C++, which doesn't have any dependency. However, Flow is written in c# so at least he need to depend on the .Net Runtime. .Net 2.0 has been abandoned by microsoft for a long time. If a program written in .Net 2 is small, it just means microsoft have shipped the required framework in windows. Flow is large because it includes a full .Net 6 runtimes inside.

If it requires you to install the runtime, it's my bad that it doesn't work as expected, and I will try again.

Though I am pointing out this, I agree with you that Flow is too large as a launcher. However, we just don't have good solution toward it.

Feel free to provide any possible solution or create pull request toward reducing our app size.

**Reply by jjw24** [reply]
Hey how's it going? 

Thought I explain a bit more as well.

>Now it asks me to install .NET Framework 6. Why would a launcher need .NET 6? What special components are there, that cannot be found anywhere else? I am not going to install .NET 6 just for this. This could be done with no .NET framework at all, or could be done with .NET 2.0 and still be compatible with all Windows versions, form XP to 11.

It requires .Net 6 framework is because this particular build you are trying is built on 6. Usually you may require to install the framework runtime in order run any programs built on .Net 5 or 6, because these frameworks are usually not pre-installed on your computer. This is the same for every program built with the .Net framework. You didnt use to install the framework separately for some .Net programs because they would have been built on older frameworks i.e. .Net framework 3.8.1 or 4.8.6 etc, which come shipped with Windows. Programs built on C++ or Python such as WinRAR or BleachBit will have different requirements.

This is the exact problem Flow Launcher's release package try to solve- not having user to install the runtime framework separately. This is done by packaging the app to include the runtime files, and these runtime framework files allow Flow to run on any machine Win7 and up without any additional installations. Basically you can move the app across different Windows devices as portable and use it straight away, without having to be admin to install anything. The downside however is what you have mentioned, size is large, because it packs all the framework runtime files that are needed to run this app on different OS versions. So essentially, the app itself is small but when you pack in these files, it's the size that you see now. And yes, there is work trying to reduce the size, so fingers crossed the size will be better.

**Comment by jjw24** [comment]
Flow will no longer launch on system start up with this release
https://github.com/Flow-Launcher/Flow.Launcher/pull/840

**Comment by rsmyst** [comment]
I have found a bug with the text typing part. here's a screenshot of it.
![image](https://user-images.githubusercontent.com/30620767/144456421-da9d8eb6-56fa-4385-a78f-659009701672.png)

It just sort of overlaps with older text when it moves past the box limit. other wise works great. im using everything if that helps with anything.
[2021-12-02.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/7643300/2021-12-02.txt)
log

Also, the overlap only seems to occur by the end of typing. i.e. until i finish typing it doesn't overlap. only happens about ~0.5 seconds after i finish typing. like its drawing over the typed part with something else.

**Reply by taooceros** [reply]
It's an existing bug. https://github.com/Flow-Launcher/Flow.Launcher/issues/153
Caused by the auto suggestion

**Reply by onesounds** [reply]
https://github.com/Flow-Launcher/Flow.Launcher/pull/854 Fixing it.

**Reply by Garulf** [reply]
Fixed in 1.9.0 https://github.com/Flow-Launcher/Flow.Launcher/pull/860

**Comment by rsmyst** [comment]
Flow Launcher version: 1.8.3
OS Version: Microsoft Windows NT 10.0.19043.0
IntPtr Length: 8
x64: True

Python Path: 
Date: 12/03/2021 11:33:53
Exception:
System.IO.InvalidDataException: Fancy Emoji|Traceback (most recent call last):
  File "C:\Users\Lenovo\Downloads\FlowLauncher\app-1.8.3\UserData\Plugins\Fancy Emoji-1.0.1\plugin.py", line 5, in <module>
    from flowlauncher import FlowLauncher
ModuleNotFoundError: No module named 'flowlauncher'

   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 258
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 272
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.QueryAsync(Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 291
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass22_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 192
--- End of stack trace from previous location ---
   at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 40
   at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 211
   at Flow.Launcher.ViewModel.MainViewModel.<>c__DisplayClass133_0.<<QueryResults>g__QueryTask|2>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 599
--- End of stack trace from previous location ---
   at Flow.Launcher.ViewModel.MainViewModel.QueryResults() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 600
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

just faced this error. 

[2021-12-03.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/7647908/2021-12-03.txt)

**Reply by jjw24** [reply]
this is coming from the plugin, need to get the plugin owner to fix it. Could you please post an issue over there.

**Comment by broadsword-MO** [comment]
Wow, Flow Launcher v1.9 is really looking good. You all sure have been busy. I appreciate the new Windows Settings search plugin. Very convenient. The layout is much improved and the UI is quite nice. I especially like the new Plugin Store; much better than having a separate web page for it. Could I suggest a little word or symbol on a plugin to show if the plugin has already been installed? Sure wish I could help you guys with this app, but I'm still learning JavaScript right now (through freeCodeCamp btw).
